pam/integration-tests/helpers_test.go

```diff
@@ -617,3 +617,21 @@ func useOldDatabaseEnv(t *testing.T, oldDB string) []string {
 
 	return []string{fmt.Sprintf("AUTHD_INTEGRATIONTESTS_OLD_DB_DIR=%s", oldDBDir)}
 }
+
+// syncBuffer is a mutex-protected buffer to avoid data races.
```

Comment:
Maybe you can just get back `safeBuffer` from b71940da9e881ed7b1b64db5e18742e116209c07

Reply:
The only difference is the RWMutex usage that imho makes sense

Reply:
agreed, changed to a RWMutex